packages/amplify-graphql-default-value-transformer/src/graphql-default-value-transformer.ts

```diff
@@ -72,10 +73,34 @@ const validateDefaultValueType = (ctx: TransformerSchemaVisitStepContextProvider
   }
 };
 
+const validateNotPrimaryKey = (field: FieldDefinitionNode): void => {
+  const isPrimaryKeyField =
+    field.directives!.find((dir) => dir.name.value === 'primaryKey') ||
+    (getBaseType(field.type) === 'ID' && field.type.kind === Kind.NON_NULL_TYPE && field.name.value === 'id');
+
+  if (isPrimaryKeyField) {
+    throw new InvalidDirectiveError('The @default directive may not be applied to primaryKey fields.');
+  }
+};
+
+const validateNotCompositeKeyMember = (config: DefaultValueDirectiveConfiguration): void => {
+  const objectDirectives = config.object.fields?.flatMap((f) => f.directives);
+  const primaryKeyDirective = objectDirectives?.find((dir) => dir?.name.value === 'primaryKey');
```

Comment:
nit: can we refactor this check to see if the field is a primary key into it's own method that can be re-used between this method and the one above?